dojo/tools/acunetix/parser.py

```diff
@@ -21,146 +22,218 @@ def get_label_for_scan_types(self, scan_type):
         return "Acunetix Scanner"
 
     def get_description_for_scan_types(self, scan_type):
-        return "XML format"
-
-    def get_findings(self, xml_output, test):
-        root = parse(xml_output).getroot()
+        return "Acunetix Scanner in XML format or Acunetix 360 Scanner in JSON format"
 
+    def get_findings(self, filename, test):
         dupes = dict()
-        for scan in root.findall("Scan"):
-            start_url = scan.findtext("StartURL")
-            if ":" not in start_url:
-                start_url = "//" + start_url
-            # get report date
-            if scan.findtext("StartTime") and "" != scan.findtext("StartTime"):
-                report_date = dateutil.parser.parse(
-                    scan.findtext("StartTime")
-                ).date()
-
-            for item in scan.findall("ReportItems/ReportItem"):
+        if '.xml' in str(filename):
```

Comment:
Could you separate the code in each of these blocks (`.xml` vs `.json`) into separate helper methods so it's easier to see which bits relate to which parser, something like `parse_acunetix_xml` and `parse_acunetix360_json` or similar? That way we have all the context for what each bit is meant to handle without tracking the whitespace through ~200 lines.

Reply:
Good idea, I can do that.

Reply:
Done @cneill